src/stdlib_bitsets.fypp

```diff
@@ -0,0 +1,2126 @@
+#:include "common.fypp"
+module stdlib_bitsets
+!! Implements zero based bitsets of size up to `huge(0_int32)`.
+!! The current code uses 64 bit integers to store the bits and uses all 64 bits.
+!! The code assumes two's complement integers, and treats negative integers as
+!! having the sign bit set.
+
+    use, intrinsic ::            &
+        iso_fortran_env, only:   &
+            bits_kind  => int32, &
+            block_kind => int64, &
+            error_unit,          &
+            int8,                &
+            int16,               &
+            int32,               &
+            int64,               &
+            dp => real64
+
+    implicit none
+
+    private
+
+    integer, parameter ::                    &
+        block_size  = bit_size(0_block_kind)
+
+    integer(block_kind), private, parameter :: all_zeros  = 0_block_kind
+    integer(block_kind), private, parameter :: all_ones   = not(all_zeros)
+
+    character(*), parameter, private :: module_name = "STDLIB_BITSETS"
+    integer, parameter, private :: &
```

Comment:
```suggestion
    integer(block_kind),  parameter :: all_zeros  = 0_block_kind
    integer(block_kind),  parameter :: all_ones   = not(all_zeros)

    character(*), parameter :: module_name = "STDLIB_BITSETS"
    integer, parameter :: &
```
`private` is not needed because the default in the module is `private`